src/coreclr/jit/compiler.h

```diff
@@ -8818,7 +8842,11 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     unsigned int maxSIMDStructBytes()
     {
 #if defined(FEATURE_HW_INTRINSICS) && defined(TARGET_XARCH)
-        if (compOpportunisticallyDependsOn(InstructionSet_AVX))
+        if (compOpportunisticallyDependsOn(InstructionSet_AVX512F))
+        {
+            return ZMM_REGSIZE_BYTES;
+        }
```

Comment:
Similar comment to the above in that we're now "paying" for this check always.

Moving it under the `if (compOpportunisticallyDependsOn(InstructionSet_AVX))` path will normalize YMM/ZMM to be 2 checks and XMM to be 1 check

Reply:
Done